### PR TITLE
Add include template function

### PR DESCRIPTION
**Pull request by twpayne**
As [suggested by](https://github.com/twpayne/chezmoi/issues/787#issuecomment-646823638) @Legion2.

This adds a template function `include` that includes the contents of a file relative to the source directory.

Notes:
* The literal contents of the file are included, the file is not interpreted as a template.
* If you want to include a template, then you need to put the file in `.chezmoitemplates` and use the `template` function.

@Legion2 I'm not sure whether we should add this there are a number of downsides:
* The `include`d file has to be added to `.chezmoiignore` or start with a `.` to prevent chezmoi from creating a file in the target state of the same name.
* The `include`d file cannot be a template itself.

In short, `include` does not add anything that cannot already be achieved with `.chezmoitemplates`, with the exception of allowing the `include`d file to be outside the `.chezmoitemplates` directory.

What are your thoughts on this?

### CODE REVIEW

**Comment by Legion2** [comment]
I currently have the setup 
`.chezmoiignore`
```
{{ if ne .chezmoi.os "windows" }}
*_windows
{{ end }}
{{ if ne .chezmoi.os "linux" }}
*_linux
{{ end }}
```

`dot_bashrc_linux`
...
`dot_bashrc_windows`
...


And then have symlink templates to link the correct os version file. For example
`symlink_dot_bashrc.tmpl`
`.bashrc_{{ .chezmoi.os }}`

But this creates two files in my home dir. The `.bashrc_linux` and the symlink `.bashrc` pointing to `.bashrc_linux`.
When using include this does only generate one file im the home dir.

Using the `.chezmoitemplates` directory looks like a good alternative, but then the files are not at the correct position in the git repository.

**Comment by twpayne** [comment]
Thanks for the feedback, so it looks like this is a useful addition. Let's go with it :)

**Comment by Legion2** [comment]
I think this new feature should be at least mentioned in the [How To guide](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#use-completely-separate-config-files-on-different-machines) for "Use completely separate config files on different machines".

**Comment by twpayne** [comment]
Totally agree! I'm planning a re-work of the templates documentation. I'll open an issue for this.
